WordPress/gutenberg
# ToolsPanel: Use Grid component to handle layout

Related:
- https://github.com/WordPress/gutenberg/pull/35415
- https://github.com/WordPress/gutenberg/issues/35058

Follow-up:
- https://github.com/WordPress/gutenberg/pull/35911

## Description

This PR refactors the `ToolsPanel` to leverage the `Grid` component to handle its layout internally.

By adopting the `Grid` component to manage the `ToolsPanel` layout, the panel will be better positioned to expose any custom grid props we desire in future. It will also remove the reliance on the currently hard-coded `single-column` class name.

## How has this been tested?

Manually. 

1. Ensured the layout and behaviour of the `ToolsPanel` in the storybook examples
2. Ensured the `ToolsPanel` displays block support panels and controls correctly.
    - Typography panel will not apply single column span until styles are tweaked in a follow-up
3. Adjusted various panels' and controls' props to play with custom column layouts.

#### Testing instructions

1. Checkout this branch and fire up the storybook: `npm run storybook:dev`
2. In the page that opens, select the [ToolsPanel > Default](http://localhost:50240/?path=/story/components-experimental-toolspanel--default) example from the left sidebar
3. Ensure the panel is laid out correctly and behaves
4. Switch to the block editor, create a post, add a cover block and select it
5. Ensure that the "Dimensions" panel in the inspector controls displays appropriately

## Screenshots 

<img width="294" alt="Screen Shot 2021-10-25 at 3 17 27 pm" src="https://user-images.githubusercontent.com/60436221/138638454-c937ca02-656e-4f90-9063-1fdb46bdbac4.png">

## Types of changes
New feature.

The removal of the reliance on the `single-column` class is a breaking change although the component is still experimental and the only adoption of the `ToolsPanel` that has merged is the "Dimensions" panel that does not use `single-column`. 

**_The Typography panel has now been switched to the ToolsPanel as well however that change hasn't been included in a release yet_**

## Checklist:
